src/commands/get/index.ts

```diff
@@ -49,19 +56,41 @@ export default class FlagsmithGet extends Command {
     }
 
     const output = flags.output
+    const entity = flags.entity
+    const isDocument = entity === 'document'
+
+    if (isDocument && !environment?.startsWith('ser.')) {
+      this.error('In order to fetch the environment document you need to provide a server-side SDK token.')
+      return
+    }
+
     outputString += `, outputing to ${output}.`
     this.log(outputString)
 
-    await flagsmith.init({
-      environmentID: environment,
-      fetch: fetch,
-      api: api,
-      identity: identity,
-    })
-    if (flags.pretty) {
-      fs.writeFileSync(output, JSON.stringify(flagsmith.getState(), null, 2))
+    if (environment?.startsWith('ser.')) {
+      fetch(`${api || 'https://edge.api.flagsmith.com/api/v1/'}environment-document/flags/`, {
+        headers: {
+          'x-environment-key': environment,
+        },
+      }).then(res => res.json()).then(res => {
+        if (flags.pretty) {
+          fs.writeFileSync(output, JSON.stringify(res, null, 2))
+        } else {
+          fs.writeFileSync(output, JSON.stringify(res))
+        }
+      })
     } else {
```

Comment:
I'd argue maybe this should be an `else if` and then we can have an `else` that throws some `'Entity not recognised'` error.

Reply:
Validation is built into the configuration

![image](https://github.com/Flagsmith/flagsmith-cli/assets/8608314/24a55685-286f-4325-b143-295a192c9e3c)